shoop/admin/modules/orders/json_order_creator.py

```diff
@@ -339,3 +295,102 @@ def update_order_from_state(self, state, order_to_update, modified_by=None):
+    creator = OrderCreator
+
+    def initialize_source_from_state(self, state, creator, ip_address, save, order_to_update=None):
+        source = super(JsonSalesOrderCreator, self).initialize_source_from_state(
+            state, creator, ip_address, save, order_to_update)
+        customer_data = state.pop("customer", None)
+        billing_address_data = customer_data.pop("billingAddress", {})
+        shipping_address_data = (
+            billing_address_data
+            if customer_data.pop("shipToBillingAddress", False)
+            else customer_data.pop("shippingAddress", {}))
+        is_company = customer_data.pop("isCompany", False)
+        save_address = customer_data.pop("saveAddress", False)
+        customer = self._get_customer(customer_data, billing_address_data, is_company, save)
+        if not customer:
+            return
+
+        billing_address = MutableAddress.from_data(billing_address_data)
+        shipping_address = MutableAddress.from_data(shipping_address_data)
+        if save:
+            billing_address.save()
+            shipping_address.save()
+
+        if save and save_address:
+            customer.default_billing_address = billing_address
+            customer.default_shipping_address = shipping_address
+            customer.save()
+
+        methods_data = state.pop("methods", None) or {}
+        shipping_method = methods_data.pop("shippingMethod")
+        if not shipping_method:
+            self.add_error(ValidationError(_("Please select shipping method."), code="no_shipping_method"))
+
+        payment_method = methods_data.pop("paymentMethod")
+        if not payment_method:
+            self.add_error(ValidationError(_("Please select payment method."), code="no_payment_method"))
+
+        if self.errors:
+            return
+
+        source.update(
+            creator=creator,
+            ip_address=ip_address,
+            customer=customer,
+            billing_address=billing_address,
+            shipping_address=shipping_address,
+            status=OrderStatus.objects.get_default_initial(),
+            shipping_method=self.safe_get_first(ShippingMethod, pk=shipping_method.get("id")),
+            payment_method=self.safe_get_first(PaymentMethod, pk=payment_method.get("id"))
+        )
+        return source
+
+
+class JsonPurchaseOrderCreator(JsonOrderCreator):
+    source = PurchaseOrderSource
+    creator = PurchaseOrderCreator
+
+    def _get_manufacturer(self, manufacturer_data, save):
+        manufacturer = self.safe_get_first(Manufacturer, pk=manufacturer_data.get("id")) if manufacturer_data else None
+        if not manufacturer:
+            if not manufacturer_data.get('name', None):
+                self.add_error(
+                    ValidationError(_("Name is required for manufacturer"), code="no_name")
+                )
+                return
+            manufacturer = Manufacturer(name=manufacturer_data['name'], url=manufacturer_data.get('url', None))
```

Comment:
Do not create new manufacturers while creatinf purchase orders. Since you are not adding new products for manufacturer you shouldn't add the manufacturer's either.